contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -85,13 +103,47 @@ abstract contract TeleporterUpgradeable is ITeleporterReceiver {
         _setMinTeleporterVersion(version);
     }
 
+    /**
+     * @dev Pauses a Teleporter address from interacting with this contract.
+     * After pausing a Teleporter address, it will no longer be able to deliver messages
+     * to this contract, and this contract will not send messages through that Teleporter address.
+     * The address does not need to be registered with the Teleporter registry.
+     * Emits a {TeleporterAddressPaused} event if successfully paused.
+     * Requirements:
+     *
+     * - `msg.sender` must have Teleporter upgrade access.
+     * - `teleporterAddress` is not already paused.
+     */
+    function pauseTeleporterAddress(address teleporterAddress) public virtual {
+        _checkTeleporterUpgradeAccess();
+        require(
+            teleporterAddress != address(0),
+            "TeleporterUpgradeable: zero Teleporter address"
+        );
+        require(
+            !isTeleporterAddressPaused(teleporterAddress),
+            "TeleporterUpgradeable: address already paused"
+        );
```

Comment:
Is there any harm in allowing this function to be called for an already paused address?

Reply:
nope there's no harm, I was just thinking as a caller it'd be nice to know if the address was already paused, vs just calling pause again for a noop. Plus the user can save gas from duplicate calls.

Reply:
Cool, just wanted to make sure there wasn't some behavior I was misunderstanding. I'm cool with keeping it as is.